### ISSUE TITLE
Combining distributed caching and autoscaling leads to inconsistency

### ISSUE THREAD

**Issue by mortenmj** [buildbuddy-io/buildbuddy-helm#23]
If distributed caching is enabled, a StatefulSet is created for buildbuddy-app-enterprise instead of the Deployment that is created otherwise. However, the HorizontalPodAutoscaler created when autoscaling is enabled has the following scaleTargetRef, which will not match the created resource:

`
    apiVersion: apps/v1
    kind: Deployment
    name: buildbuddy-buildbuddy-enterprise
`

**Comment by siggisim** [comment]
Hey @mortenmj - we should definitely fix this, but it may be for the best (or remote this auto-scaler for now).

The executor deployment is fine to autoscale very aggressively up and down. The apps however are a bit more sensitive to rapid autoscaling. Scaling up apps up or down quickly can be hard for the app replication to keep up with in short period of time. For that reason, we recommend autoscaling executors - but keeping a fixed pool of apps that are able to handle your maximum load.

We're working on a big project to change the way we do replication between the apps that will make app autoscaling less brittle. I will keep this issue updated with progress once this is live.

**Comment by tylerwilliams** [comment]
Yeah I'd be OK disabling this autoscaler if the apps are going to be in a StatefulSet. We def do not want to be autoscaling then.

(TBD if we'll want to autoscale apps in a raft world; we'll see!)
